# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/generic/ViewportFrame.cpp
# Commit: 00af531b9d94
# Full Hash: 00af531b9d943a7a536494ad8e9256c327383aab
# Author: Matt Woodrow <mwoodrow@mozilla.com>
# Date: 2021-05-16 09:17:48
# Regressor Bug: 1540737
# File Overlap Count: 1
# Description:
#   Bug 1540737 - Don't allow constructing an AutoBuildingDisplayList with a new frame without requiring explicit rects in the new coordinate space. r=miko
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D114675
# ==============================================================================

diff -r eeba14b5d89a -r 00af531b9d94 layout/generic/ViewportFrame.cpp
--- a/layout/generic/ViewportFrame.cpp	Sat May 15 22:38:45 2021 +0000
+++ b/layout/generic/ViewportFrame.cpp	Sat May 15 22:38:45 2021 +0000
@@ -229,8 +229,10 @@
   if (topLayerList.IsEmpty()) {
     return nullptr;
   }
-  nsDisplayListBuilder::AutoBuildingDisplayList buildingDisplayList(aBuilder,
-                                                                    this);
+  nsPoint offset = aBuilder->GetCurrentFrame()->GetOffsetTo(this);
+  nsDisplayListBuilder::AutoBuildingDisplayList buildingDisplayList(
+      aBuilder, this, aBuilder->GetVisibleRect() + offset,
+      aBuilder->GetDirtyRect() + offset);
   // Wrap the whole top layer in a single item with maximum z-index,
   // and append it at the very end, so that it stays at the topmost.
   nsDisplayWrapList* wrapList = MakeDisplayItemWithIndex<nsDisplayWrapList>(